stage #4/module #2. REST API Advanced/rest_api_advanced.md

```diff
@@ -18,105 +18,124 @@
+ORM stands for **Object Relational Mapping**. It’s a bit of an abstract concept – but basically it’s a technique that allows us to query and change data from the database in an object oriented way. ORMs provide a high-level abstraction upon a relational database that allows a developer to write Java code instead of SQL to create, read, update and delete data and schemas in their database. Developers can use the programming language they are comfortable with to work with a database instead of writing SQL statements or stored procedures.\
+A JPA **(Java Persistence API)** is a specification of Java which is used to access, manage, and persist data between Java object and relational database. It is considered as a standard approach for Object Relational Mapping. JPA can be seen as a bridge between object-oriented domain models and relational database systems. Being a specification, JPA doesn't perform any operation by itself. Thus, it requires implementation. So, ORM tools like Hibernate, TopLink, and iBatis implements JPA specifications for data persistence.\
+A **transaction** usually means a sequence of information exchange and related work (such as database updating) that is treated as a unit for the purposes of satisfying a request and for ensuring database integrity. For a transaction to be completed and database changes to made permanent, a transaction has to be completed in its entirety.
 
-##### Application restrictions
+##### Application requirements
 
-1. Hibernate specific features.
-2. Spring Data
+1. Gradle, latest version. Multi-module project.
+2. Application packages root: com.mjc.school.
+3. Hibernate should be used as a JPA implementation for data access.
+4. Spring Framework & Spring Boot, the latest version.
+5. Spring Transaction should be used in all necessary areas of the application.
+6. Java Code Convention is mandatory.
+7. Application restriction: Spring Data
 
 # Demo
 ## Practical part
 
-1. Generate for a demo at least 
-     - 1000 users
-     - 1000 tags
-     - 10’000 gift certificates (should be linked with tags and users)
-All values should look like more -or-less meaningful: random words, but not random letters 
-2. Demonstrate API using Postman tool (prepare for demo Postman collection with APIs)  
-3. (Optional) Build & run application using command line
+1. Application should be deployed before demo.
+2. Generate for a demo at least
+   - 1000 users
+   - 1000 tags
+   - 10’000 gift certificates (should be linked with tags and users)
+     All values should look like more -or-less meaningful: random words, but not random letters
+3. Demonstrate API using Postman tool. Prepare for demo Postman collection with APIs.
+4. (Optional) Build & run application using command line
 
 ## Theoretical part
 
-Mentee should be able to answer questions during demo session.
+Mentee should be able to answer theoretical and practical questions during demo session.
 
 ## Extra Materials
 
```

Comment:
Maybe it worth provide some Swagger link as soon as we recommend to use it for documentation

Reply:
added a link